src/liquidator.ts

```diff
@@ -0,0 +1,647 @@
+                pubkey: new PublicKey(this.clusterConfig.borrowing.accounts.oracleMappings), // oracleMappings
+                isWritable: false,
+                isSigner: false
+            },
+        ].concat(...Object.keys(CollateralTokenActive).filter(token => isNaN(parseInt(token))).map(token => {
+            return {
+                pubkey: new PublicKey(this.clusterConfig.borrowing.accounts.pyth[token.toLowerCase() + 'PriceInfo']), // pythPriceInfo
+                isWritable: false,
+                isSigner: false
+            };
+        })).concat(...[
+            {
+                pubkey: TOKEN_PROGRAM_ID, // tokenProgram
+                isWritable: false,
+                isSigner: false
+            },
+            {
+                pubkey: SYSVAR_CLOCK_PUBKEY, // clock
+                isWritable: false,
+                isSigner: false
+            }
+        ]) as AccountMeta[];
+        return new TransactionInstruction({
+            data: this.hubbleProgram.coder.instruction.encode('tryLiquidate', args),
+            programId: this.hubbleProgram.programId,
+            keys: keys
+        });
+    }
+
+
+    // clear the liquidation queue
+    // async clearLiquidationQueue() : Promise<void> {
+
+    //     // filter the liquidation queue for events which are pending collection
+    //     const filteredLiquidationQueue = this.liquidationsQueue.events.filter(e => e.status.toNumber() === EventStatus.PendingCollection);
+    //     // loop through each liquidation event
+    //     const txs = flat(await Promise.all(filteredLiquidationQueue.map((pendingEvent, index) => {
+    //         return new Promise((resolve) => { 
+    //             // 1 second timeout per liquidation event
+    //             setTimeout(async () => {
+    //                 resolve(await this.clearLiquidationGains());
+    //             }, 1000 * index);
+    //         });
+    //     })));
+    //     txs.forEach(tx => {
+    //         console.log('https://solscan.io/tx/'+tx+(this.cluster === 'devnet' ?  '?cluster=devnet' : ''));
+    //     });
+    // }
+
+    // clear liquidation gains
+    async clearLiquidationGains() : Promise<Array<string>> {
+        const blockhash = (await this.hubbleProgram.provider.connection.getRecentBlockhash()).blockhash;
+        return flat(await Promise.all(Object.keys(CollateralTokenActive).filter(token => isNaN(parseInt(token))).map(async (token) => {
+            let tx = new Transaction();
+            tx.add(this.clearLiquidationGainsIXMap.get(token));
+            tx.feePayer = this.hubbleProgram.provider.wallet.publicKey;
+            tx.recentBlockhash = blockhash;
+            tx = await this.hubbleProgram.provider.wallet.signTransaction(tx);
+            return await (this.hubbleProgram.provider.connection as TpuConnection).sendRawTransaction(tx.serialize());
+        })));
+    }
+
+    // get the pending debt of a user
+    async getPendingDebt(user : UserMetadata) : Promise<Decimal> {
```

Comment:
any reason you are not using our sdk and recreating this?

Reply:
Initially I had just used BN then realized the price was getting it's decimals truncated, then I added the Decimal... I'd like to move to using the sdk!

Reply:
It works better this way, the UserMetadata is stored on the polling-accounts-fetcher accounts map. I'd rather not have duplicate data on the bot (UserMetadata BN) and (UserMetadata Decimal)